Bottom: 29ad88f5f8a8118889256cc8f2e07db96b0c4bb2
Top:    1d14aa4ee55a5a06b5ff5fb738b58701159f3cb4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-17 03:05:59 +0100

Add debug catch to triangle handling

If for any reason we are about to run off the end of the
triangle array, deliberately segfault so gdb can catch where.


---

diff --git a/src/hid/common/hidgl.h b/src/hid/common/hidgl.h
index fc1f37a..615e72b 100644
--- a/src/hid/common/hidgl.h
+++ b/src/hid/common/hidgl.h
@@ -43,6 +43,13 @@ hidgl_add_triangle_3D (triangle_buffer *buffer,
                        GLfloat x2, GLfloat y2, GLfloat z2,
                        GLfloat x3, GLfloat y3, GLfloat z3)
 {
+  if (buffer->triangle_count == TRIANGLE_ARRAY_SIZE)
+    {
+      fprintf (stderr, "hidgl_add_triangle_3D: "
+               "Attempt to add past end of array. Will now segfault.\n");
+      *(char *)0 = 0;
+    }
+
   buffer->triangle_array [buffer->coord_comp_count++] = x1;
   buffer->triangle_array [buffer->coord_comp_count++] = y1;
   buffer->triangle_array [buffer->coord_comp_count++] = z1;
